packages/coordinator/ts/sessionKeys/sessionKeys.service.ts

```diff
@@ -0,0 +1,129 @@
+import { Injectable, Logger } from "@nestjs/common";
+import { deserializePermissionAccount } from "@zerodev/permissions";
+import { toECDSASigner } from "@zerodev/permissions/signers";
+import { createKernelAccountClient, KernelAccountClient, KernelSmartAccount } from "@zerodev/sdk";
+import { KERNEL_V3_1 } from "@zerodev/sdk/constants";
+import { ENTRYPOINT_ADDRESS_V07 } from "permissionless";
+import { ENTRYPOINT_ADDRESS_V07_TYPE } from "permissionless/types";
+import { type Chain, createPublicClient, http, HttpTransport, Transport } from "viem";
+import { generatePrivateKey, privateKeyToAccount } from "viem/accounts";
+
+import { ErrorCodes } from "../common";
+import { genPimlicoRPCUrl } from "../common/accountAbstraction";
+import { CryptoService } from "../crypto/crypto.service";
+import { FileService } from "../file/file.service";
+
+import { IGenerateSessionKeyReturn } from "./types";
+
+/**
+ * SessionKeysService is responsible for generating and managing session keys.
+ */
+@Injectable()
+export class SessionKeysService {
+  /**
+   * Logger
+   */
+  private readonly logger: Logger;
+
+  /**
+   * Create a new instance of DeployerService
+   * @param fileService
+   */
+  constructor(
+    private readonly cryptoService: CryptoService,
+    private readonly fileService: FileService,
+  ) {
+    this.fileService = fileService;
+    this.logger = new Logger(SessionKeysService.name);
+  }
+
+  /**
+   * Generate a session key
+   *
+   * @returns session key address
+   */
+  generateSessionKey(): IGenerateSessionKeyReturn {
+    const sessionPrivateKey = generatePrivateKey();
+
+    const sessionKeySigner = toECDSASigner({
+      signer: privateKeyToAccount(sessionPrivateKey),
+    });
+
+    const sessionKeyAddress = sessionKeySigner.account.address;
+
+    // save the key
+    this.fileService.storeSessionKey(sessionPrivateKey, sessionKeyAddress);
```

Comment:
Maybe it makes sense to use redis instead of saving keys to file?

Reply:
maybe as a future improvement?

Reply:
Sure, let's do it next. It shouldn't take much time.